pkg/controller/registry/reconciler/configmap.go

```diff
@@ -354,6 +361,8 @@ func (c *ConfigMapRegistryReconciler) ensureRoleBinding(source configMapCatalogS
 
 func (c *ConfigMapRegistryReconciler) ensurePod(source configMapCatalogSourceDecorator, overwrite bool) error {
 	pod := source.Pod(c.Image)
+	c.MuPod.Lock()
```

Comment:
It's not clear to me that adding a mutex solves the problem at hand. The queues that are used key on name/namespace and should never allow the same resource to be processed by more than one worker concurrently.

IMO, an alternative suggestion for the behavior you're seeing is that the resource was processed more than once, serially, before the cache that backs the `Lister` -- which you've swapped with a client call above --  can be updated. This could have lead to cache misses, and the resource being recreated.

I would feel much more comfortable if you could produce a test, unit or e2e, that fails for the original code but passes with these changes.

Reply:
Thanks for review.  I saw 2 pods were created so I guessed the cause of it.  I'll look into more detail and capture what is happening.  I'm not sure if I can create a test to reproduce consistently but I'll try it anyway.